### PR TITLE
Move reaction to models/issues/

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

File: models/db/map.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package db
+
+// KeysInt64 returns keys slice for a map with int64 key
```
Comment (6543) [inline models/db/map.go]: this helper should move itno golbal utils package ... and can use generics asap golang v1.18 is min requirement

Comment (lunny) [reply models/db/map.go]: Done.

**Review by 6543 — APPROVED** [review]

File: models/issues/reaction.go [hunk 2]
```diff
@@ -105,32 +168,31 @@ func findReactions(e db.Engine, opts FindReactionsOptions) ([]*Reaction, int64,
...
-	}
-	if opts.Comment != nil {
-		reaction.CommentID = opts.Comment.ID
-		findOpts.CommentID = opts.Comment.ID
+		UserID:    opts.DoerID,
 	}
 
-	existingR, _, err := findReactions(e, findOpts)
+	reaction.CommentID = opts.CommentID
+	findOpts.CommentID = opts.CommentID
```
Comment (wxiaoguang) [inline models/issues/reaction.go]: Move these two assignments into the `&Reaction{` and `FindReactionsOptions{` ?

And the old code `CommentID: -1, // reaction to issue only` seems strange.

Comment (lunny) [reply models/issues/reaction.go]: done.

File: models/issues/reaction.go [hunk 3]
```diff
@@ -105,32 +168,29 @@ func findReactions(e db.Engine, opts FindReactionsOptions) ([]*Reaction, int64,
...
+		Type:      opts.Type,
+		UserID:    opts.DoerID,
+		IssueID:   opts.IssueID,
+		CommentID: opts.CommentID,
 	}
 	findOpts := FindReactionsOptions{
-		IssueID:   opts.Issue.ID,
-		CommentID: -1, // reaction to issue only
+		IssueID:   opts.IssueID,
+		CommentID: opts.CommentID, // reaction to issue only
```
Comment (wxiaoguang) [inline models/issues/reaction.go]:
```suggestion
		CommentID: opts.CommentID,
```

What does ` // reaction to issue only` really mean?

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**